fastlane/lib/fastlane/plugins/plugin_manager.rb

```diff
@@ -54,8 +54,10 @@ def self.to_gem_name(plugin_name)
     # Returns an array of gems that are added to the Gemfile or Pluginfile
     def available_gems
       return [] unless gemfile_path
-      dsl = Bundler::Dsl.evaluate(gemfile_path, nil, true)
-      return dsl.dependencies.map(&:name)
+      # Bundler's `dependencies` method returns all gems included in Gemfile, ignoring enabled Bundler groups.
+      # To get only gems from enabled groups, we need to combine with `specs`, which does include only gems from enabled groups.
+      definition = Bundler.definition
```

Comment:
```suggestion
      definition = Bundler::Dsl.evaluate(gemfile_path, nil, true)
```
I think this can still use `Bundle::Dsl.evaluate` as this returns a definition. We need to this so that we can pass a `gemfile_path`during tests

Reply:
@joshdholtz thanks for review!

It's true that `Dsl.evaluate` returns a definition, but it does not perform actual dependency resolution, so `specs` method returns `nil` on that. Instead, Bundler.definition gives you currently active environment which includes dependency resolution.

Maybe we can force returned `definition` to do resolution again using `resolve_remotely!` or `resolve_with_cache!` methods, but there is no guarantee that it will be the same as currently active bundler environment.


I believe that this implementation behaves correctly, but I just couldn't find an easy way to change Bundler.definition behavior for unit tests.

Maybe we can extract inline `Bundler.definition` call to a method, and stub that method for unit tests to return a resolved bundler definition from specific gemfile we want in unit tests?

Reply:
It's been a few weeks since I looked at this but it didn't look like it was returning the correct/expected gems. Either that or _fastlane_ isn't doing the correct things with what is being returned. I will run this again and see if I can provide a better analysis of what's happening and what's expected 🤔